04-onion-routing.md

```diff
@@ -630,7 +631,7 @@ The origin node:
   - once the return message has been decrypted:
     - SHOULD store a copy of the message.
     - SHOULD continue decrypting, until the loop has been repeated 20 times.
-    - SHOULD use constant `ammag` and `um` keys to deobfuscate [FIXME: is this correct? was 'obfuscate'] the route length.
+    - SHOULD use constant `ammag` and `um` keys to obfuscate the route length. [FIXME: should this be indented under 'once return message has been decrypted?']
```

Comment:
No, this is correct.  If we have a route length 7, we only have 7 ammag and um keys.  But we really need to do the decryption 20 times (to avoid timing attacks which may detect that we finished early), so we use dummy ones.

Reply:
Ah tricky ok, thanks.